src/components/Requirements/IncompleteSelfCheck.vue

```diff
@@ -101,6 +101,8 @@ export default defineComponent({
             return;
           }
 
+          /* TODO @bshen fix .allowCourseDoubleCounting flag
+             we should allow the constraint violation to be broken, i.e. don't return early */
```

Comment:
What does this refer to?

Reply:
The usage of `.allowCourseDoubleCounting` in the following lines.

```
// There are 2 ways we can add the course to the requirement without double counting violations:
// 1. This requirement allows double counting.
// 2. All the already matched requirements allow double counting.
// If both don't hold, we cannot add.
```

We should change this to allow adding regardless of whether or not the requirement allows double counting, and resolve the constraint violations later (basically remove the early return). This fix should wait until after most of the frontend is implemented.

Reply:
Can you make an item for this haha